redbot/core/commands/dpy_reimplements.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+import inspect
+import functools
+from typing import (
+    TypeVar,
+    Callable,
+    Awaitable,
+    Coroutine,
+    Union,
+    Type,
+    TYPE_CHECKING,
+    List,
+    Any,
+    Generator,
+    Protocol,
+    overload,
+)
+
+import discord
+from discord.ext import commands as dpy_commands
+
+# So much of this can be stripped right back out with proper stubs.
+from discord.ext.commands import (
+    check as check,
+    guild_only as guild_only,
+    dm_only as dm_only,
+    is_nsfw as is_nsfw,
+    has_role as has_role,
+    has_any_role as has_any_role,
+    bot_has_role as bot_has_role,
+    bot_has_any_role as bot_has_any_role,
+    cooldown as cooldown,
+)
```

Comment:
Conditional import is fine here. It may be ugly, but the point is for this to work for our users and devs, if a little bit more ugly importing in this module is needed for it, that's fine.